nestjs/src/opportunities/opportunities.controller.ts

```diff
@@ -79,6 +105,24 @@ export class OpportunitiesController {
     return new ConsentDto(data);
   }
 
+  @Post('/status')
+  @ApiOperation({ operationId: 'updateStatus' })
+  @ApiOkResponse({ type: StatusDto })
+  @UseGuards(DefaultGuard)
+  public async updateStatus(@Req() req: CurrentRequest) {
```

Comment:
IMO looks not good: `update status`, but inside we do some implicit prolongation. I am not aware about the `opportunity` feature, but probably smth  like `prolong` would be better